processing/src/test/java/io/druid/query/extraction/BucketExtractionFnTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ *  Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ *  or more contributor license agreements. See the NOTICE file
+ *  distributed with this work for additional information
+ *  regarding copyright ownership. Metamarkets licenses this file
+ *  to you under the Apache License, Version 2.0 (the
+ *  "License"); you may not use this file except in compliance
+ *  with the License. You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ */
+
+package io.druid.query.extraction;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.druid.jackson.DefaultObjectMapper;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class BucketExtractionFnTest
+{
+  private static final double DELTA = 0.0000001;
+
+  @Test
+  public void testApply()
+  {
+    BucketExtractionFn extractionFn1 = new BucketExtractionFn(100.0, 0.5);
+    Assert.assertEquals("1200.5", extractionFn1.apply("1234.99"));
+    Assert.assertEquals("0.5", extractionFn1.apply("1"));
+    Assert.assertEquals("0.5", extractionFn1.apply("100"));
+    Assert.assertEquals("500.5", extractionFn1.apply(501));
+    Assert.assertEquals("-399.5", extractionFn1.apply("-325"));
```

Comment:
Would be great to see a test for input that is non numeric, specifically `""` and `"blah blah"` also `"1e2"` would be an interesting test cases